reactor-core/src/test/java/reactor/core/publisher/FluxWindowTimeoutTest.java

```diff
@@ -469,4 +470,32 @@ public void scanMainSubscriber() {
 		Assertions.assertThat(test.scan(Scannable.Attr.CANCELLED))
 		          .isTrue();
 	}
+
+	@Test
+	public void sourceError() {
+		TestPublisher<Integer> source = TestPublisher.create();
+
+		StepVerifier.create(source.flux()
+		                          .windowTimeout(2, Duration.ofSeconds(1), true)
+		                          .flatMap(Flux::materialize)
+		            )
+					.then(() -> source.next(1))
+					.expectNext(Signal.next(1))
+		            .then(() -> source.error(
+				            new IllegalStateException("expected failure")))
+		            // failure observed by window
+		            .expectNextMatches(signalSourceErrorMessage("expected failure"))
+		            // failure observed by main Flux
+		            .expectErrorMessage("expected failure")
+		            .verify();
+
+		source.assertNoSubscribers();
+	}
+
+	private <T> Predicate<? super Signal<T>> signalSourceErrorMessage(String expectedMessage) {
```

Comment:
I duplicated this method as some window tests don't share a common base – namely `FluxWindowBoundaryTest`, `FluxWindowTimeoutTest`, and `FluxWindowWhenTest`. `FluxWindowPredicateTest`, and `FluxWindowTest` on the other hand, do both inherit from `FluxOperatorTest`. Perhaps a more thorough inspection should follow, but I considered it out of scope for this change.